.github/workflows/ci.yml

```diff
@@ -0,0 +1,49 @@
+on:
+  pull_request:
+    branches: [main]
+
+  push:
+    branches: [main]
+
+defaults:
+  run:
+    working-directory: popgetter-browser
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install pnpm
+        uses: pnpm/action-setup@v4
+        with:
+          version: 9
+
+      - name: Install wasm-pack
+        uses: jetli/wasm-pack-action@v0.4.0
+
+      - name: Cache WASM build
+        uses: actions/cache@v3
+        with:
+          path: rust_backend/target
+          key: rust_backend
+
+      - name: Install dependencies for all packages
+        run: |
+          cd web/
+          pnpm install
+
+      - name: Build web app
+        run: |
+          cd web
+          pnpm rust-release
+          pnpm build
+
+      - name: Publish
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: ./web/dist/
```

Comment:
Drive-by comment cause I'm excited to try this (without having to build myself): this should be `./web/build/` for sveltekit

Reply:
Amazing, was just trying to figure this out, thanks so much!

Reply:
For future debugging:
1) https://github.com/Urban-Analytics-Technology-Platform/popgetter-browser/tree/gh-pages is empty, why?
2) I went to the action logs at https://github.com/Urban-Analytics-Technology-Platform/popgetter-browser/actions/runs/11708675395/job/32610993647
3) 
![image](https://github.com/user-attachments/assets/29585b26-0781-4c8f-96c4-bf9e1216b527)
4) 
![image](https://github.com/user-attachments/assets/6a78cca4-190a-4dd0-8e9a-35e59f7a2cda)

I've spent hours debugging GH actions/pages nonsense before. :(  The other random trick that can sometimes help is to check if something like `wasm-pack` is generating a `.gitingore` file in a directory you're trying to publish, because the action to commit to a `gh-pages` branch will pay attention to that

Reply:
Thanks a lot for the suggestions @dabreegster, they were very helpful. With the empty `gh-pages`, the build was not being committed in the publish step and it turns out `publish_dir` does not combine with the specified default `working-directory` I specified in the workflow. So updating to `publish_dir: popgetter-browser/web/build` fixed it.